src/openvic-simulation/map/MapDefinition.cpp

```diff
@@ -637,6 +658,183 @@ bool MapDefinition::load_map_images(fs::path const& province_path, fs::path cons
+	static constexpr uint8_t RIVER_SIZE_9 = 10;
+	static constexpr uint8_t RIVER_SIZE_10 = 11;
+
+	uint8_t const* river_data = rivers_bmp.get_pixel_data().data();
+
+	/** Generating River Segments - Unoptimised & Unprofiled
+		1. check pixels up, right, down, and left from last_segment_end for a colour <12
+		2. add first point
+		3. set size of segment based on color value at first point
+		4. loop, adding adjacent points until the colour value changes (to make sure we don't backtrack, last_segment_direction provides a pixel to automatically ignore)
+				last_segment_direction:
+				0 -> start, ignore nothing
+				1 -> ignore up
+				2 -> ignore down
+				3 -> ignore left
+				4 -> ignore right
+		5. if the colour value changes to MERGE_COLOUR, add the point & finish the segment
+		6. if there is no further point, finish the segment
+		7. if the colour value changes to a different river size (>1 && <12), recursively call this function on the next segment
+	*/
+	const std::function<void(ivec2_t, uint8_t, river_t&)> next_segment = [&river_data, &rivers_bmp, &next_segment](ivec2_t last_segment_end, uint8_t last_segment_direction, river_t& river) {
+		size_t idx = last_segment_end.x + last_segment_end.y * rivers_bmp.get_width();
+
+		std::vector<ivec2_t> points;
+		uint8_t direction;
+
+		// check pixel above
+		if (last_segment_end.y > 0 && last_segment_direction != 1) { // check for bounds & ignore direction
+			if (river_data[idx - rivers_bmp.get_width()] < 12) {
+				points.push_back({ last_segment_end.x, last_segment_end.y - 1 });
+				direction = 2;
+			}	
+		}
+		// check pixel to right
+		if (last_segment_end.x < rivers_bmp.get_width() - 1 && last_segment_direction != 4) {
+			if (river_data[idx + 1] < 12) {
+				points.push_back({ last_segment_end.x + 1, last_segment_end.y });
+				direction = 3;
+			}
+		}
+		// check pixel below
+		if (last_segment_end.y < rivers_bmp.get_height() - 1 && last_segment_direction != 2) {
+			if (river_data[idx + rivers_bmp.get_width()] < 12) {
+				points.push_back({ last_segment_end.x, last_segment_end.y + 1 });
+				direction = 1;
+			}
+		}
+		// check pixel to left
+		if (last_segment_end.x > 0 && last_segment_direction != 3) {
+			if (river_data[idx - 1] < 12) {
+				points.push_back({ last_segment_end.x - 1, last_segment_end.y });
+				direction = 4;
+			}
+		}
+
+		uint8_t size = river_data[points.front().x + points.front().y * rivers_bmp.get_width()] - 1; // size of river from 1 - 10 determined by colour
+
+		bool river_complete = false;
+		ivec2_t new_point;
+
+		size_t limit = 0; // stops infinite loop
+
+		while (true) {
+			limit++;
+			if (limit == 4096) {
+				Logger::error("River segment starting at (", points.front().x, ", ", points.front().y, ") is longer than limit 4096, check for misplaced pixels or other definition errors!");
+				river_complete = true;
+				break;
+			}
+
+			idx = points.back().x + points.back().y * rivers_bmp.get_width();
+
+			ivec2_t merge_location;
+			bool merge;
```

Comment:
This variable could be randomly true or false if not explicitly initialised (with MSVS at least), so it needs to be `bool merge = false;`